Project/backend/codebase/graph_creator/router.py

```diff
@@ -0,0 +1,168 @@
+import logging
+import os
+import uuid
+
+from fastapi import APIRouter, Depends
+from fastapi import UploadFile, File, HTTPException
+
+from graph_creator.dao.graph_job_dao import GraphJobDAO
+from graph_creator.schemas.graph_job import GraphJobCreate
+
+router = APIRouter()
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+# Endpoint for uploading PDF documents
+@router.post("/upload/")
+async def upload_pdf(
+    file: UploadFile = File(...), graph_job_dao: GraphJobDAO = Depends()
+):
+    """
+    Uploads a PDF document.
+
+    Args:
+        file (UploadFile): PDF document to be uploaded.
+        graph_job_dao (GraphJobDAO):
+
+    Returns:
+        dict: A dictionary containing the filename and status.
+            id (uuid.UUID): Id of the uploaded file
+            filename (str): Name of the uploaded file.
+            location (str): Location of the uploaded file.
+            status (str): Status message of the file.
+
+    Raises:
+        HTTPException: If the uploaded file type is not valid (not PDF).
+    """
+
+    # Check if the uploaded file type is correct
+    if file.content_type != "application/pdf":
+        raise HTTPException(status_code=400, detail="Uploaded file is not a PDF.")
+
+    # Define the directory for saving files
+    documents_directory = os.path.join(
+        os.path.dirname(os.path.dirname(os.path.abspath(__file__))), ".media/documents"
+    )
+    if not os.path.exists(documents_directory):
+        os.makedirs(documents_directory)
+    logger.info(documents_directory)
+
+    # Save file
+    file_path = os.path.join(documents_directory, file.filename)
+    with open(file_path, "wb") as f:
+        f.write(file.file.read())
+    logger.info(file_path)
```

Comment:
I think this part does not have the correct behavior. If the user uploads a file with the same name but different content it will overwrite an already saved file. I think if a file exists with the same name we should append some string (e.g current datetime) so the location becomes unique and a new file is saved.

Reply:
adjusted it so that it raises error, if file with the same name is uploaded, so it doesn't overwrite the existing one. POs suggested we leave out uploading file with the same name for now.